`TestDisableSchedulingServiceFallback` is flaky

## Flaky Test

### Which jobs are failing
```
    server_test.go:109: 
        	Error Trace:	/home/runner/work/pd/pd/tests/integrations/mcs/scheduling/server_test.go:109
        	Error:      	Should NOT be empty, but was 
        	Test:       	TestServerTestSuite/TestAllocIDAfterLeaderChange
```
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x42186f0]

goroutine 450604 [running]:
github.com/tikv/pd/tests/integrations/mcs/scheduling.(*serverTestSuite).TestDisableSchedulingServiceFallback.func1()
	/home/runner/work/pd/pd/tests/integrations/mcs/scheduling/server_test.go:262 +0x50
github.com/stretchr/testify/assert.Eventually.func1()
	/home/runner/go/pkg/mod/github.com/stretchr/testify@v1.9.0/assert/assertions.go:1902 +0x34
created by github.com/stretchr/testify/assert.Eventually in goroutine 450613
	/home/runner/go/pkg/mod/github.com/stretchr/testify@v1.9.0/assert/assertions.go:1902 +0x3d6
```

### CI link
https://github.com/tikv/pd/actions/runs/12367690342/job/34516396487?pr=8887

### Reason for failure (if possible)


### Anything else

--- a/tests/integrations/mcs/scheduling/server_test.go
+++ b/tests/integrations/mcs/scheduling/server_test.go
@@ -304,8 +304,9 @@ func (suite *serverTestSuite) TestDisableSchedulingServiceFallback() {
 
 	// PD will execute scheduling jobs since there is no scheduling server.
 	testutil.Eventually(re, func() bool {
-		re.NotNil(suite.pdLeader.GetServer())
-		re.NotNil(suite.pdLeader.GetServer().GetRaftCluster())
+		if suite.pdLeader.GetServer() == nil || suite.pdLeader.GetServer().GetRaftCluster() == nil {
+			return false
+		}
 		return suite.pdLeader.GetServer().GetRaftCluster().IsSchedulingControllerRunning()
 	})
 	leaderServer := suite.pdLeader.GetServer()
@@ -335,7 +336,15 @@ func (suite *serverTestSuite) TestDisableSchedulingServiceFallback() {
 	})
 	// Scheduling server is responsible for executing scheduling jobs.
 	testutil.Eventually(re, func() bool {
-		return tc.GetPrimaryServer().GetCluster().IsBackgroundJobsRunning()
+		primaryServer := tc.GetPrimaryServer()
+		if primaryServer == nil {
+			return false
+		}
+		cluster := primaryServer.GetCluster()
+		if cluster == nil {
+			return false
+		}
+		return cluster.IsBackgroundJobsRunning()
 	})
 	// Disable scheduling service fallback and stop scheduling server. PD won't execute scheduling jobs again.
 	conf.EnableSchedulingFallback = false

